catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -123,18 +145,168 @@ public Fileset createFileset(
       String storageLocation,
       Map<String, String> properties)
       throws NoSuchSchemaException, FilesetAlreadyExistsException {
-    throw new UnsupportedOperationException();
+    try {
+      if (store.exists(ident, Entity.EntityType.FILESET)) {
+        throw new FilesetAlreadyExistsException("Fileset " + ident + " already exists");
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to check if fileset " + ident + " exists", ioe);
+    }
+
+    SchemaEntity schemaEntity;
+    NameIdentifier schemaIdent = NameIdentifier.of(ident.namespace().levels());
+    try {
+      schemaEntity = store.get(schemaIdent, Entity.EntityType.SCHEMA, SchemaEntity.class);
+    } catch (NoSuchEntityException exception) {
+      throw new NoSuchSchemaException("Schema " + schemaIdent + " does not exist", exception);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to load schema " + schemaIdent, ioe);
+    }
+
+    // For external fileset, the storageLocation must be set.
+    if (type == Fileset.Type.EXTERNAL && StringUtils.isBlank(storageLocation)) {
+      throw new IllegalArgumentException(
+          "Storage location must be set for external fileset " + ident);
+    }
+
+    // Either catalog property "location", or schema property "location", or storageLocation must be
+    // set for managed fileset.
+    Path schemaPath = getSchemaPath(schemaIdent.name(), schemaEntity.properties());
+    if (schemaPath == null && StringUtils.isBlank(storageLocation)) {
+      throw new IllegalArgumentException(
+          "Storage location must be set for fileset "
+              + ident
+              + " when it's catalog and schema location are not set");
+    }
+
+    // The specified storageLocation will take precedence over the calculated one.
+    Path filesetPath =
+        StringUtils.isNotBlank(storageLocation)
+            ? new Path(storageLocation)
+            : new Path(schemaPath, ident.name());
+    try {
+      FileSystem fs = filesetPath.getFileSystem(hadoopConf);
+      if (!fs.exists(filesetPath)) {
+        if (!fs.mkdirs(filesetPath)) {
+          throw new RuntimeException(
+              "Failed to create fileset " + ident + " location " + filesetPath);
+        }
+
+        LOG.info("Created fileset {} location {}", ident, filesetPath);
+      } else {
+        LOG.info("Fileset {} manages the existing location {}", ident, filesetPath);
+      }
+
+    } catch (IOException ioe) {
+      throw new RuntimeException(
```

Comment:
Optional: 
Maybe we can use `UncheckedIOException` here.

Reply:
We typically use `RuntimeException` everywhere, I think it is not so necessary to change.

Reply:
Maybe we can create a good first issue for this. WDYT?